core/ai_worker.go

```diff
@@ -0,0 +1,1054 @@
+package core
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"path"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/golang/glog"
+	"github.com/livepeer/ai-worker/worker"
+	"github.com/livepeer/go-livepeer/clog"
+	"github.com/livepeer/go-livepeer/common"
+	"github.com/livepeer/go-livepeer/monitor"
+	"github.com/livepeer/go-livepeer/net"
+	"github.com/livepeer/go-tools/drivers"
+	"github.com/livepeer/lpms/ffmpeg"
+)
+
+var ErrRemoteWorkerTimeout = errors.New("Remote worker took too long")
+var ErrNoCompatibleWorkersAvailable = errors.New("no workers can process job requested")
+var ErrNoWorkersAvailable = errors.New("no workers available")
+
+// TODO: consider making this dynamic for each pipeline
+var aiWorkerResultsTimeout = 10 * time.Minute
+var aiWorkerRequestTimeout = 15 * time.Minute
+
+type RemoteAIWorker struct {
+	manager      *RemoteAIWorkerManager
+	stream       net.AIWorker_RegisterAIWorkerServer
+	capabilities *Capabilities
+	eof          chan struct{}
+	addr         string
+}
+
+func (rw *RemoteAIWorker) done() {
+	// select so we don't block indefinitely if there's no listener
+	select {
+	case rw.eof <- struct{}{}:
+	default:
+	}
+}
+
+type RemoteAIWorkerManager struct {
```

Comment:
Wrong file structure

Reply:
Most AI related things are in separate files right?

Can you help me understand why using `ai_worker.go` files does not help keep development on AI and transcoding from causing issues for anyone developing on one or the other?